src/scripts/pokemons/PokemonFactory.ts

```diff
@@ -37,6 +37,15 @@ class PokemonFactory {
         return new BattlePokemon(name, id, basePokemon.type1, basePokemon.type2, maxHealth, route * 2, catchRate, exp, money, shiny);
     }
 
+    public static validRoute(route: number, region: GameConstants.Region): boolean {
```

Comment:
Really nice

Reply:
I was thinking of moving this into MapHelper maybe, it feels out of place in PokemonFactory.